karapace/protobuf/enum_element.py

```diff
@@ -0,0 +1,61 @@
+# Ported from square/wire:
+# wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/internal/parser/EnumElement.kt
+from itertools import chain
+from karapace.protobuf.compare_result import CompareResult, Modification
+from karapace.protobuf.compare_type_storage import CompareTypes
+from karapace.protobuf.enum_constant_element import EnumConstantElement
+from karapace.protobuf.location import Location
+from karapace.protobuf.type_element import TypeElement
+from karapace.protobuf.utils import append_documentation, append_indented
+
+
+class EnumElement(TypeElement):
+    def __init__(
+        self, location: Location, name: str, documentation: str = "", options: list = None, constants: list = None
+    ) -> None:
+        # Enums do not allow nested type declarations.
+        super().__init__(location, name, documentation, options or [], [])
+        self.constants = constants or []
+
+    def to_schema(self) -> str:
+        result: list = []
+        append_documentation(result, self.documentation)
+        result.append(f"enum {self.name} {{")
+
+        if self.options or self.constants:
+            result.append("\n")
+
+        if self.options:
+            for option in self.options:
+                append_indented(result, option.to_schema_declaration())
+
+        if self.constants:
+            for constant in self.constants:
+                append_indented(result, constant.to_schema())
+
+        result.append("}\n")
+        return "".join(result)
+
+    def compare(self, other: 'EnumElement', result: CompareResult, types: CompareTypes) -> None:
```

Comment:
we have one instance of result because it contains common data for all compare functions in the chain. This instance we create before comparison of two schemas and at final it contains a data which required for last calculation of compatibility.